### DISCUSSION TITLE
Amazing app , Ability to interact with open ai platform.

### DISCUSSION THREAD

**Discussion by rachitpant** [alagrede/znote-app#81] [Q&A]
This is an amazing app , i just tried it yesterday for the first time and was hooked onto until very late past midnight.  I was using [StackEdit+](https://stackedit.net/app#) which also has chatgpt integration , but with the programmatic capability of this app , this looks so much more powerful.

Here is what i am trying.
Will it be possible for Znotes to interact with the open Ai/ Chatgpt platform with/without a token and write 3 articles on a topic. ( eg. most important psychology research ) and then generate 3 docs ( markdowns).

**Comment by alagrede** [comment]
Hi @rachitpant,
Thank you for for your feedback 😊. I'm not sure to understand your need.
Just to be sure we're talking about the same thing: Znote has an OpenAI integration accessible with `Ctrl+H`. You need to add your OpenAI token in the settings to work with it.
If you are an expert, you can also customize OpenAI settings to fit your use-case.
Have you seen this feature? Or are you talking about a completely different thing?
Thank you for your clarifications 🙏
Anthony

**Comment by rachitpant** [comment]
This is more like a request to help me with this.  I thinking of having a javascript/python script that converses with chatgpt and and saves the each response to a document locally. eg. first request would be to find 30 interesting research papers on marketing. and then 30 addtional requests in a loop asking chat gpt to summarize each of those articles and save it into a document. 

After getting each response , znote could bring up a prompt asking to add the response to the current document , or create a new document with the response ( with an option to - always do this ).

I think with the programmatic capabilities of znote it has a great potential and having something like this as a ` recipe` could make it very popular among bloggers / article writers.

Also to answer your question , I did try asking chatgpt a query with `Ctrl+H. ` , but i always got a 429 response code. i have added the key, this response code is probably because i am using the free version of chat gpt.
Thanks for all you are doing. @alagrede

**Comment by alagrede** [comment]
Hi @rachitpant,
Yes, you are right, you need a paid account to avoid these 429 errors. The `Ctrl+H` box gives you a history of recent and favorite orders. (Very useful!)

But, thank you for your explanation. I understand and it is actually a good idea! 😊
I've attached a quick example for you, of how you might go about manipulating OpenAI. I hope this will be useful to you!
I will add some OpenAI manipulation examples in the recipes list. 
[openai-test.md](https://github.com/alagrede/znote-app/files/11404951/openai-test.md)

Best regards,
Anthony

**Comment by rachitpant** [comment]
Thank you so much for this and for Znote. I look forward for to try those examples in recipes.
